demo/src/moveit_task_constructor_demo.cpp

```diff
@@ -0,0 +1,128 @@
+/*********************************************************************
+ * BSD 3-Clause License
+ *
+ * Copyright (c) 2019 PickNik LLC.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Henning Kayser, Simon Goldstein
+   Desc:   A demo to show MoveIt Task Constructor in action
+*/
+
+// ROS
+#include <ros/ros.h>
+
+// MTC pick/place demo implementation
+#include <moveit_task_constructor_demo/pick_place_task.h>
+
+#include <geometry_msgs/Pose.h>
+#include <moveit/planning_scene_interface/planning_scene_interface.h>
+
+void spawnTable()
+{
+  ros::Duration(1.0).sleep();
+  ros::NodeHandle pnh("~");
+  std::string table_name = pnh.param<std::string>("table_name", "table");
+  std::string surface_frame = pnh.param<std::string>("table_surface_frame", "world");
```

Comment:
I added both subframes and I renamed the variables to refrence frames rather than surfaces.

Reply:
@henningkayser The sub-frames added break Travis because they use an updated moveit_msgs that must be built from source and 2/3 Travis tests don't build from source. Is this something worth reverting?

Reply:
Actually, only one test breaks because of the subframes.
The other one failed for seemingly [unrelated reasons](https://github.com/ros-planning/moveit_ci/issues/85).

This patch brings up the question whether we need MTC to build on top of upstream ROS melodic, or whether we are willing to require MoveIt's `master` branch instead.
I would be ok with requesting the `master` branch, although it seems weird to change this version dependency for a tutorial instead of core features (which might follow soon).

@rhaschke

Reply:
I prefer to keep MTC compatible with MoveIt Kinetic, Melodic, and master.
Moving the demo code to `moveit_tutorials` will resolve this dependency problem as well.

Reply:
I put the code in this package to follow the model set by [moveit_grasps](https://ros-planning.github.io/moveit_tutorials/doc/moveit_grasps/moveit_grasps_tutorial.html). If this should be code in the tutorial it may also be valuable to move the other as well.

Reply:
All other code examples of the tutorials are located in the tutorials repo. I think, moveit_grasps already followed a non-standard approach there.
So, yes, I advocate moving all tutorial related demos into the tutorials repo.